### ISSUE TITLE
using editor bar on mobile keeps closing keyboard

### ISSUE THREAD

**Issue by kulizivo** [PKM-er/obsidian-editing-toolbar#143]
Basically, every time I try to use the toolbar it ends up closing the keyboard on everything except the highlighter. This means every other command will end up closing the keyboard and only leaving the toolbar active.

**Comment by codey-badger** [comment]
Are u still having this issue?
If yes don't forget to leave the devices info ^^

**Comment by kulizivo** [comment]
@codey-badger Yes
Android 13
Redmi note 10Pro
128 - 6ram
Minimal theme

**Comment by ChasKane** [comment]
same; I ran into this seemingly unavoidable issue when trying to develop my own toolbar -- it seems that all clicks/taps on anything other than text, even buttons inserted into the markdown, cause the keyboard to disappear.

**Comment by codey-badger** [comment]
@ChasKane yeah it's quite unfortunate and annoying tbh. Today tested the plugin and sometimes the toolbar itself disappears from top or has the erratic behavior with the keyboard. 

As a heavy mobile user, my unsolicited advice is:
1. Sticking to the core plugins and functions as much as possible (I use **_Slash commands_** extensively). 
2.  Refining the workflow using the command `configure mobile toolbar` and ***Hotkeys*** (option in settings). 

For more functionality you could try [Note Toolbar Plugin](https://github.com/chrisgurney/obsidian-note-toolbar) and see how it goes. Although I'm comfortable with the core mobile toolbar + [Commander Plugin](https://github.com/phibr0/obsidian-commander) + [QuickAdd Plugin](https://github.com/chhoumann/quickadd) here and there. 

Another recommendation I'm currently testing is [Text format plugin](https://github.com/Benature/obsidian-text-format). 

Finally [Settings Search Plugin](https://github.com/anpigon/obsidian-book-search-plugin) to make customization easier. 

Hope that helps!

**Comment by kulizivo** [comment]
@ChasKane  & @codey-badger  try popkit on mobile

**Comment by ChasKane** [comment]
o.0 ahhhhh @kulizivo is your account ok? popkit looks like spam.

**Comment by ChasKane** [comment]
@codey-badger `configure mobile toolbar` is a gem and so is the reminder about slash commands, thank you! Otherwise, l think we're in the same boat. I have a dream of a colorful, dynamic, animation-ladened toolbar with similar features to this plugin but a smoother UI and better customization potential, but with Obsidian being as opaque as it is, I have to relent that my energy is better spent elsewhere.

**Comment by kulizivo** [comment]
@ChasKane yes its really bad on mobile but on desktop it works much better

**Comment by ChasKane** [comment]
@codey-badger the erratic behavior we discussed seems to be addressed by adding `event.preventDefault()` to both the `click` and `mousedown` event listeners set on each command button. I'll have a prototype of my own toolbar up soon

```React JSX
onClick={(e) => {
  global.app.commands.executeCommandById(command.id);
  e.preventDefault();
}}
onMouseDown={(e) => e.preventDefault()}
```

**Comment by kulizivo** [comment]
@ChasKane thanks for the edit. tell me when your version of the toolbar is up... ill be happy to test drive it for you

**Comment by ChasKane** [comment]
@kulizivo it's looking p good. drag and drop buttons, customize commands, colors, etc. It feels *really good* to use, even if it's a hair buggy. I'm still working on it but if you wanna give it a shot, `git clone` the repo into your plugins folder, run `npm run dev` to compile the typescript to `main.js`, and have a blast ^.^

https://github.com/ChasKane/gay-toolbar

**Comment by kulizivo** [comment]
@ChasKane hey i checked this out, interesting name. But in terms of use custmization its really good. buttons work just fine and it definitely works a dream... though it is still a little buggy and and should have adjustments not to appear on pc if the user only wants it mobile... outside of that amazing plugin... by the way as just a suggestion, you could also make it possible to use an aimage for the background and also add transparency sliders for the buttons as well

**Comment by ChasKane** [comment]
So glad you like it! I already have both the desktop/mobile issue and the button transparency issue on my list. As for backgrounds, I think in the latest versions you can do that with the customcss option.

I'm working on a bug rn that's causing the text cursor to go under the toolbar. v annoying. but the solution seems to have simplified the plugin immensely. I'll keep you posted ^.^

**Comment by ChasKane** [comment]
For desktop/mobile, I think I'll leave it up to users to disable it on desktop and not on mobile via the Sync feature (you can sync the plugins list but not the list of which ones of that list are active). What do you think?

**Comment by kulizivo** [comment]
well its okay if that's the only option but in my case I use third party syncing so ill have to look for a work around

**Comment by ChasKane** [comment]
Lmk what you find. I might be able to implement it in the next update. I'll have to experiment with loading the settings synchronously before render -- if that works, I can just disable the plugin if a "use on desktop" setting is off. I'll keep you posted.

**Comment by ChasKane** [comment]
@kulizivo I added the mobile-only feature, will be available in the second release

**Comment by kulizivo** [comment]
okay update me when you release it since i try out the app on a separate device

**Comment by ChasKane** [comment]
@kulizivo it's up, just search `gay toolbar` in the community plugins! How's it working for ya?

**Comment by kulizivo** [comment]
kul... ill check

**Comment by cumany** [comment]
Thanks for the feedback, this has been fixed in version 3.x.

**Comment by codey-badger** [comment]
> Thanks for the feedback, this has been fixed in version 3.x.

Hi @cumany , I noticed that when clicking a "submenu" containing more commands it still hides keyboard. It would be possible to fix it for next update? Thanks in advance.
